src/app/core/services/issue.service.ts

```diff
@@ -334,22 +335,67 @@ export class IssueService {
     // const issuesAPICallsByFilter = filters.map(filter => this.githubService.fetchIssuesGraphql(filter));
     return forkJoin(issuesAPICallsByFilter).pipe(
       map((issuesByFilter: [][]) => {
+        const fetchedIssueIds: Array<Number> = [];
+
         for (const issues of issuesByFilter) {
           for (const issue of issues) {
+            fetchedIssueIds.push(this.createIssueModel(issue).id);
             this.createAndSaveIssueModel(issue);
           }
         }
+
+        const outdatedIssueIds: Array<Number> = this.getOutdatedIssueIds(fetchedIssueIds);
+        this.deleteIssuesFromLocalStore(outdatedIssueIds);
+
         return Object.values(this.issues);
       })
     );
   }
 
+
   private createAndSaveIssueModel(githubIssue: GithubIssue): boolean {
     const issue = this.createIssueModel(githubIssue);
     this.updateLocalStore(issue);
     return true;
   }
 
+  private deleteIssuesFromLocalStore(ids: Array<Number>): void {
+    ids.forEach((id: number) => {
+      this.getIssue(id).subscribe(issue => this.deleteFromLocalStore(issue));
+    });
+  }
+
+  /**
+   * Returns an array of outdated issue ids by comparing the ids of the recently
+   * fetched issues with the current issue ids in the local store
+   */
+  private getOutdatedIssueIds(fetchedIssueIds: Array<Number>): Array<Number> {
+    const result = [];
+
+    /*
+      Ignore for first fetch or ignore if there is no fetch result
+
+      We also have to ignore for no fetch result as the cache might return a
+      304 reponse with no differences in issues, resulting in the fetchIssueIds
+      to be empty
+    */
+    if (this.issues === undefined || !fetchedIssueIds.length) {
+      return result;
+    }
+
+    const fetchedIssueIdsSet = new Set<Number>(fetchedIssueIds);
+
+    const originalIssueIds = Object.keys(this.issues);
+
+    originalIssueIds.forEach((issueId: string) => {
+      if (!fetchedIssueIdsSet.has(+issueId)) {
+        result.push(+issueId);
+      }
+    });
+
+    return result;
```

Comment:
```suggestion
    const result = Object.keys(this.issues).map(x => parseInt(x)).filter(issueId => !fetchedIssueIdsSet.has(issueId));
    return result;
```
Let's use pure functions here

Reply:
Hey @dingyuchen, sorry I didn't see this! Will change according to your suggestion!